sdks/java/core/src/main/java/org/apache/beam/sdk/util/RowJson.java

```diff
@@ -185,18 +186,25 @@ private RowJsonDeserializer(Schema schema) {
       this.schema = schema;
     }
 
+    public RowJsonDeserializer allowMissingFields(Boolean allowMissing){
+      this.allowMissingFields = allowMissing;
+      return this;
+  }
+
     @Override
     public Row deserialize(JsonParser jsonParser, DeserializationContext deserializationContext)
         throws IOException {
 
       // Parse and convert the root object to Row as if it's a nested field with name 'root'
       return (Row)
           extractJsonNodeValue(
-              FieldValue.of("root", FieldType.row(schema), jsonParser.readValueAsTree()));
+              FieldValue.of("root", FieldType.row(schema), jsonParser.readValueAsTree(), allowMissingFields));
     }
 
+  
+
     private static Object extractJsonNodeValue(FieldValue fieldValue) {
-      if (!fieldValue.isJsonValuePresent()) {
```

Comment:
I initially thought about this, then changed to putting it into the FieldValue because extractJsonNodeValue was static. Now that you mention it, I believe it is probably neater to just check this.allowMissing fields and change the method to non-static.

Reply:
Ah yeah I didn't notice it was static. I think making it non-static is the right call, thanks!